code/modules/projectiles/projectile.dm

```diff
@@ -1131,8 +1146,11 @@ So if we are on the 32th absolute pixel coordinate we are on tile 1, but if we a
 			ammo.do_at_max_range(src)
 			end_of_movement = TRUE
 			break
-		if(HAS_TRAIT(next_turf, TRAIT_TURF_FREEZE_BULLET))
+		if(HAS_TRAIT(next_turf, TRAIT_TURF_BULLET_MANIPULATION))
+			SEND_SIGNAL(next_turf, COMSIG_TURF_PROJECTILE_MANIPULATED, src)
 			QDEL_LIST_IN(laser_effects, 1)
+			if(HAS_TRAIT_FROM(next_turf, TRAIT_TURF_BULLET_MANIPULATION, PORTAL_TRAIT))
+				return
```

Comment:
whats with these snowflake seeming trait checks

Reply:
Less performance cost than having two traits to check.

We check if something weird is happening to the bullet, and if yes, we are looking for the source so we know what we shoudl do with it